clients/java/src/main/java/io/zeebe/client/impl/BearerTokenCredentialsProvider.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.impl;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.grpc.Metadata;
+import io.grpc.Metadata.Key;
+import io.zeebe.client.CredentialsProvider;
+import java.io.File;
+import java.io.IOException;
+
+public class BearerTokenCredentialsProvider implements CredentialsProvider {
+  private static final Key<String> HEADER_AUTH_KEY =
+      Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
+  private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory());
+  private final String zeebeCredentialsPath;
+
+  public BearerTokenCredentialsProvider(String zeebeCredentialsPath) {
+    if (zeebeCredentialsPath == null
+        || zeebeCredentialsPath.isEmpty()
+        || !new File(zeebeCredentialsPath).exists()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Expected valid path to Zeebe credentials but '%s' is either invalid or does not point to a file.",
+              zeebeCredentialsPath));
+    }
+
+    this.zeebeCredentialsPath = zeebeCredentialsPath;
+  }
+
+  /**
+   * Adds a JSON Web Token (JWT) to the Authorization header of a gRPC call. The JWT is obtained
+   * from a provided Zeebe credentials YAML file.
+   */
+  @Override
+  public void applyCredentials(Metadata headers) {
+    try {
+      final ZeebeClientCredentials credentials = getAccessCredentials();
+
+      headers.put(
+          HEADER_AUTH_KEY,
+          String.format(
+              "%s %s", credentials.getTokenType().trim(), credentials.getAccessToken().trim()));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private ZeebeClientCredentials getAccessCredentials() throws IOException {
+    JsonNode node = MAPPER.readTree(new File(zeebeCredentialsPath));
+
+    node = node.get("endpoint").get("auth");
+    final ZeebeClientAuthInfo clientAuthInfo =
+        MAPPER.readValue(node.traverse(), new TypeReference<ZeebeClientAuthInfo>() {});
```

Comment:
We can probably save the `TypeReference` as a constant? In fact, isn't there a way to get an `ObjectReader` for a specific type and save that as our constant instead of having both the mapper and the type reference? I could be misremembering.

Reply:
> In fact, isn't there a way to get an `ObjectReader` for a specific type and save that as our constant

Yes, I could get an object reader and then just `readValue` but I wanted to skip the first level (since right now we're assuming it's just "endpoint") so I had to get a JsonNode to do that and from there this was the only way I knew of reading to an object. But there might be a better way, I'm not sure. 
The "auth" level also doesn't have anything other than the credentials at this point (I'm not sure if it'll have anything soon) so maybe I could delete ZeebeClientAuthInfo and if the zeebe token spec is update we can also update this.